### ISSUE TITLE
Sort by two fields

### ISSUE THREAD

**Issue by guilload** [quickwit-oss/quickwit#2670]
You can only sort by one field at the moment.

**Comment by fulmicoton** [comment]
Supporting 2 numerical fields might be sufficient.

**Comment by PSeitz** [comment]
How should we handle fields with mixed types?

**Comment by fulmicoton** [comment]
>  How should we handle fields with mixed types?

We should handle it well. 

We have two worlds here.
Within a segment we need performance.

The SortingFieldComputer today emits `Option<u64>` regardless of which u64-based column is used for sorting. (using the monotonic mapping. You need to modify the SortingFieldComputer to return `(Option<u64>, Option<u64>)`, or something like that that is stack allocated.

As you harvest results from a split, you will need to get the actual top K score values and be more dynamic. I think this is done today with enums.

**Comment by PSeitz** [comment]
With mixed types we could currently have bool/numeric(u64, i64, f64)
- Mixed on the same segment => We could either merge them or just pick the numeric column
- Mixed between segments (I think that's already handled)

I think mixed on the same segment is broken currently as this may accidentally sort on term ids.

**Comment by PSeitz** [comment]
```json
{"myjson": true}
{"myjson": "nice"}
{"myjson": 55}
```

This will create 3 columns. When sorting on `myjson`, we should
1. Not pick the term id column (as they are not comparable across segments)
2. Either pick the numeric column, or merge bool and numeric when fetching the values

**Comment by fulmicoton** [comment]
Ok. This is not really related to this isssue then? This is an issue even with a single field?

Today the spec we have is that we pick the first column in that order and ignore others. (This is not great)
numerical, bytes, string.
bool is not supported.

**Comment by PSeitz** [comment]
> Ok. This is not really related to this isssue then? This is an issue even with a single field?

Yes, not specific to this issue. I just thought I cover that when working on it, since I did spot some issues in the code.

#### API
What should the [REST API](https://quickwit.io/docs/main-branch/reference/rest-api#parameters-1) look like?

Currently we have the `sort_by_field` parameter and `sort_order`, while `sort_order` seems to be undocumented currently.
I think we have 3 options:

* Allow multiple fields in one parameter  `sort_by_field`  and `sort_order`: `field1,field2`
* Explicit paramters: `sort_by_field2`  and `sort_by_order2`
* Only make it available in the ES API

**Comment by guilload** [comment]
Regarding the API, I would like to propose an alternative option:

In the REST API:
Rename the `sort_by_field` to `sort_by`. We would keep using the sort "mini-dsl", now supporting at most two sort fields: `sort_by=±<field name>,±<field name> `.

In the gRPC API:
Remove the `sort_by_field` and `sort_order` parameters. Introduce a new `sort_by` parameter which holds an array of `SortField { field_name, sort_order }`:

```proto
  // Optional sort by one or more fields (limited to 2 at the moment). 
  optional SortBy sort_by = 13;
}

message SortBy {
  repeated SortField sort_fields = 1;
}

message SortField {
  string field_name = 1;
  SortOrder sort_order = 2;
}
```

**Comment by fmassot** [comment]
closed by #3614
